rskj-core/src/test/java/co/rsk/peg/NonStandardErpFederationsTest.java

```diff
@@ -337,48 +326,107 @@ void createLegacyErpRedeemScript_fromNonStandardErpBuilders() {
+            emergency0PublicKey, emergency1PublicKey, emergency2PublicKey
+        );
+        emergencyThreshold = emergencyKeys.size() / 2 + 1;
+        activationDelayValue = 52_560L;
 
+        // we should activate RSKIP284
+        when(activations.isActive(ConsensusRule.RSKIP284)).thenReturn(true);
+
+        // this should create the expected non-standard with csv unsigned be fed
         federation = createDefaultNonStandardErpFederation();
-        builder = new NonStandardErpRedeemScriptBuilderWithCsvUnsignedBE();
-        obtainedRedeemScript = builder
-            .createRedeemScriptFromKeys(defaultKeys, defaultThreshold,
+
+        ErpRedeemScriptBuilder builder = federation.getErpRedeemScriptBuilder();
+        Script obtainedRedeemScript = builder
+            .createRedeemScriptFromKeys(
+                defaultKeys, defaultThreshold,
                 emergencyKeys, emergencyThreshold,
                 activationDelayValue
             );
         assertArrayEquals(expectedRedeemScriptProgram, obtainedRedeemScript.getProgram());
+    }
 
-
-        // test NonStandardErpRedeemScriptBuilder
-        expectedRedeemScriptProgram = // this is the redeem script program from fed non-standard
+    @Test
+    void createdRedeemScriptProgramFromNonStandardErpBuilder_withRealValues_equalsRealRedeemScriptProgram_mainnet() {
+        byte[] expectedRedeemScriptProgram = // this is the redeem script program from fed non-standard
             Hex.decode("6453210208f40073a9e43b3e9103acec79767a6de9b0409749884e989960fee578012fce210225e892391625854128c5c4ea4340de0c2a70570f33db53426fc9c746597a03f421025a2f522aea776fab5241ad72f7f05918e8606676461cb6ce38265a52d4ca9ed62102afc230c2d355b1a577682b07bc2646041b5d0177af0f98395a46018da699b6da2103fb8e1d5d0392d35ca8c3656acb6193dbf392b3e89b9b7b86693f5c80f7ce858155670350cd00b27552210216c23b2ea8e4f11c3f9e22711addb1d16a93964796913830856b568cc3ea21d3210275562901dd8faae20de0a4166362a4f82188db77dbed4ca887422ea1ec185f1421034db69f2112f4fb1bb6141bf6e2bd6631f0484d0bd95b16767902c9fe219d4a6f5368ae");
 
-        federator4PublicKey = BtcECKey.fromPublicOnly(Hex.decode("03fb8e1d5d0392d35ca8c3656acb6193dbf392b3e89b9b7b86693f5c80f7ce8581"));
+        // these values belong to the non-standard fed
+        BtcECKey federator0PublicKey = BtcECKey.fromPublicOnly(Hex.decode("0208f40073a9e43b3e9103acec79767a6de9b0409749884e989960fee578012fce"));
+        BtcECKey federator1PublicKey = BtcECKey.fromPublicOnly(Hex.decode("0225e892391625854128c5c4ea4340de0c2a70570f33db53426fc9c746597a03f4"));
+        BtcECKey federator2PublicKey = BtcECKey.fromPublicOnly(Hex.decode("025a2f522aea776fab5241ad72f7f05918e8606676461cb6ce38265a52d4ca9ed6"));
+        BtcECKey federator3PublicKey = BtcECKey.fromPublicOnly(Hex.decode("02afc230c2d355b1a577682b07bc2646041b5d0177af0f98395a46018da699b6da"));
+        BtcECKey federator4PublicKey = BtcECKey.fromPublicOnly(Hex.decode("03fb8e1d5d0392d35ca8c3656acb6193dbf392b3e89b9b7b86693f5c80f7ce8581"));
```

Comment:
This same public key is being created and used in different tests. What do you think about moving it outside the method and converting it into a class instance variable?

Reply:
Yes, good idea. I will create another pr for that